public/app/plugins/datasource/prometheus/datasource.tsx

```diff
@@ -726,6 +727,34 @@ export class PrometheusDatasource
       datasource: this.getRef(),
     };
 
+    if (this.access === 'direct') {
+      const start = this.getPrometheusTime(options.range.from, false);
+      const end = this.getPrometheusTime(options.range.to, true);
+      const queryOptions = { ...options, interval: step };
+
+      const queryModel = {
+        expr,
+        interval: step,
+        refId: 'X',
+        requestId: `prom-query-${annotation.name}`,
+      };
+      const query = this.createQuery(queryModel, queryOptions, start, end);
+      return await lastValueFrom(
+        this.performTimeSeriesQuery(query, query.start, query.end).pipe(
+          filter((response: any) => isFetchSuccessResponse(response)),
```

Comment:
😢 Help requested for the type

Reply:
This one is challenging! I was unable to figure it out.

Reply:
```suggestion
          filter(isFetchSuccessResponse),
```

I think this should work. I've found this good blog post on it https://medium.com/ngconf/filtering-types-with-correct-type-inference-in-rxjs-f4edf064880d

Reply:
@ivanahuckova that is worked. But I tried that while I was working on this and it failed. I must have missed something. Thanks!